src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -719,12 +780,17 @@ private byte[] GetAsciiEncoding(byte value)
             {
                 if (!WillEncode(value))
                 {
-                    _asciiEscape[value] = s_noEscape;
-                    return s_noEscape;
+                    encoding = s_noEscape;
+                    _asciiEscape[value] = encoding;
                 }
             }
 
             return encoding;
         }
+
+        private static void ThrowArgumentException_MaxOutputCharsPerInputChar()
+        {
+            throw new ArgumentException("Argument encoder does not implement MaxOutputCharsPerInputChar correctly.");
```

Comment:
I think we generally put such exception message strings in a Strings resource file and use `SR.` to access them.

Reply:
The S.T.E.W project currently does not use string resources.

Reply:
https://github.com/dotnet/corefx/issues/40074